AmplifyPlugins/Storage/Tests/AWSS3StoragePluginTests/Service/Storage/AWSS3StorageServiceGetPreSignedURLBehaviorTests.swift

```diff
@@ -94,4 +101,54 @@ class AWSS3StorageServiceGetPreSignedURLBehaviorTests: XCTestCase {
             "getPreSignedURL(key:signingOperation:expires:) \(serviceKey ?? "") \(operation) \(String(describing: expires))"
         ])
     }
+
+    /// - Given: A storage service configured to use a AWSS3PreSignedURLBuilder
+    /// - When: Validation for a non-existence is requested
+    /// - Then: A StorageError.keyNotFound is thrown
+    func testvalidateObjectExistenceForNonExistentKey() async throws {
+        client.headObjectHandler = { _ in
+            throw HeadObjectOutputError.notFound(.init())
+        }
+        let nonExistentKey = UUID().uuidString
+        do {
+            _ = try await systemUnderTest.validateObjectExistence(serviceKey: nonExistentKey)
+            XCTFail("Expecting error but validateObjectExistence succeeded")
+        } catch StorageError.keyNotFound(let key, _, _, _) {
+            XCTAssertTrue(key.contains(nonExistentKey), "nonExistentKey: \(nonExistentKey), key: \(key)")
+        }
+    }
+
+    /// - Given: A storage service configured to use a AWSS3PreSignedURLBuilder
+    /// - When: Validation for a non-existence is requested
+    /// - Then: An SdkError.service is thrown
+    func testvalidateObjectExistenceForNonExistentKeyWithSdkServiceError() async throws {
+        client.headObjectHandler = { _ in
+            let headObjectError = HeadObjectOutputError.notFound(.init())
+            throw SdkError<HeadObjectOutputError>.service(headObjectError, HttpResponse(body: .none, statusCode: .notFound))
+        }
+        let nonExistentKey = UUID().uuidString
+        do {
+            _ = try await systemUnderTest.validateObjectExistence(serviceKey: nonExistentKey)
+            XCTFail("Expecting error but validateObjectExistence succeeded")
+        } catch StorageError.keyNotFound(let key, _, _, _) {
```

Comment:
Can we have extra assertions on `SdkError` type here to make a distinction from the `testvalidateObjectExistenceForNonExistentKey`

Reply:
actually nvm, i saw we're returning `StorageError` after mapping in the code.